test/PreAMMBatcher.spec.ts

```diff
@@ -0,0 +1,105 @@
+import {use, expect} from 'chai';
+import {Contract, utils} from 'ethers';
+import {deployContract, deployMockContract, MockProvider, solidity} from 'ethereum-waffle';
```

Comment:
Where did you pick up all this fancy new contract tooling?

Reply:
waffle is amazing. read here: https://ethereum-waffle.readthedocs.io/en/stable/mock-contract.html